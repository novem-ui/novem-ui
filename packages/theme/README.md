# `@novem-ui/theme`

> TODO: description

## Usage

```
const theme = require('@novem-ui/theme');

// TODO: DEMONSTRATE API
```
