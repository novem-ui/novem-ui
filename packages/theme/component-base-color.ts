export { ComponentBaseColorProps, ComponentBaseColorProvider, useComponentBaseColor } from './src/component-base-color'
