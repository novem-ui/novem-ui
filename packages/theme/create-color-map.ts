export { default } from './src/utils/create-color-map'
