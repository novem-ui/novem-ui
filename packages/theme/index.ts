export { default as useTheme } from './use-theme'
export { default as withTheme } from './with-theme'
export { default as createColorMap } from './create-color-map'
export { default as ThemeProvider } from './theme-provider'
export { default, Theme, BaseColor, Color, HeadingSizeValue, SpacingValue, TextSizeValue, Shadows } from './theme'
