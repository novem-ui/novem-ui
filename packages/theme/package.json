{
  "name": "@novem-ui/theme",
  "version": "0.0.2-alpha.0",
  "description": "Theme for novem-ui ",
  "author": "Leonel Vieyra <me@leonelv.com>",
  "homepage": "https://github.com/jdrets/novem-ui/tree/main/packages/theme#readme",
  "license": "ISC",
  "main": "index.js",
  "publishConfig": {
    "access": "public",
    "directory": "../../dist/theme"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/jdrets/novem-ui.git"
  },
  "scripts": {
    "test": "jest --env=jsdom --passWithNoTests",
    "build": "yarn build:esm && yarn build:types && yarn build:package-json",
    "build:esm": "../../node_modules/.bin/babel . --root-mode upward --extensions .ts,.tsx -d ../../dist/theme --source-maps",
    "build:types": "tsc -p ./tsconfig.types.json",
    "build:package-json": "../../scripts/create-package.js"
  },
  "bugs": {
    "url": "https://github.com/jdrets/novem-ui/issues"
  },
  "peerDependencies": {
    "@emotion/react": "^11.1.5",
    "@emotion/styled": "^11.1.5",
    "polished": "^4.1.3",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}
