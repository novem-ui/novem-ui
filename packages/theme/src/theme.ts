import colors from './colors'
import * as typography from './typography'
import spacing from './spacing'
import { WeightedColorKey } from '../types'

const theme = {
  colors,
  typography,
  spacing,
}

export type BaseColor = `${keyof typeof colors.palette}`
export type Color = `${keyof typeof colors.palette}.${WeightedColorKey}`
export type HeadingSizeValue = keyof typeof typography.headingFontSizes
export type SpacingValue = keyof typeof spacing
export type TextSizeValue = keyof typeof typography.textFontSizes
export type Theme = typeof theme


export default theme
