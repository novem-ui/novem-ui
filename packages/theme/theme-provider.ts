export { default } from './src/theme-provider'
