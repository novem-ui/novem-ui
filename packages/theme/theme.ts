export { default, Theme, BaseColor, Color, HeadingSizeValue, SpacingValue, TextSizeValue } from './src/theme'
