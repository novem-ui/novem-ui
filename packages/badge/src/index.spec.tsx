import React from 'react'

import theme, { withTheme } from '@novem-ui/theme'
import { render } from '@testing-library/react'

import Badge from '.'

describe('<Badge />', () => {
  test('should match snapshot (default)', () => {
    const Component = () => <Badge>Dummy text</Badge>
    const WrappedComponent = withTheme(Component, theme)
    const { container } = render(<WrappedComponent />)

    expect(container).toMatchSnapshot()
  })

  test('should match snapshot (small)', () => {
    const Component = () => <Badge size="sm">Dummy text</Badge>
    const WrappedComponent = withTheme(Component)
    const { container } = render(<WrappedComponent />)

    expect(container).toMatchSnapshot()
  })

  test('should match snapshot (quiet)', () => {
    const Component = () => <Badge hierarchy="quiet">Dummy text</Badge>
    const WrappedComponent = withTheme(Component)
    const { container } = render(<WrappedComponent />)

    expect(container).toMatchSnapshot()
  })

  test('should match snapshot (alternative color)', () => {
    const Component = () => <Badge baseColor="green">Dummy text</Badge>
    const WrappedComponent = withTheme(Component)
    const { container } = render(<WrappedComponent />)

    expect(container).toMatchSnapshot()
  })
})
