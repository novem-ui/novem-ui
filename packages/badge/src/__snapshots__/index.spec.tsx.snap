// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should match snapshot (alternative color) 1`] = `
.emotion-0 {
  background-color: #59BC6F;
  color: rgba(255,255,255,1);
  border-radius: 1rem;
  display: inline-block;
  font-size: 14px;
  font-weight: 600;
  line-height: 1.2;
  padding: 0.25rem 0.75rem;
  text-transform: uppercase;
}

<div>
  <span
    class="emotion-0 emotion-1"
  >
    Dummy text
  </span>
</div>
`;

exports[`<Badge /> should match snapshot (default) 1`] = `
.emotion-0 {
  background-color: #F1694B;
  color: rgba(255,255,255,1);
  border-radius: 1rem;
  display: inline-block;
  font-size: 14px;
  font-weight: 600;
  line-height: 1.2;
  padding: 0.25rem 0.75rem;
  text-transform: uppercase;
}

<div>
  <span
    class="emotion-0 emotion-1"
  >
    Dummy text
  </span>
</div>
`;

exports[`<Badge /> should match snapshot (quiet) 1`] = `
.emotion-0 {
  background-color: rgba(241,105,75,0.15);
  color: #F1694B;
  border-radius: 1rem;
  display: inline-block;
  font-size: 14px;
  font-weight: 600;
  line-height: 1.2;
  padding: 0.25rem 0.75rem;
  text-transform: uppercase;
}

<div>
  <span
    class="emotion-0 emotion-1"
  >
    Dummy text
  </span>
</div>
`;

exports[`<Badge /> should match snapshot (small) 1`] = `
.emotion-0 {
  background-color: #F1694B;
  color: rgba(255,255,255,1);
  border-radius: 1rem;
  display: inline-block;
  font-size: 12px;
  font-weight: 600;
  line-height: 1;
  padding: 0.125rem 0.5rem;
  text-transform: uppercase;
}

<div>
  <span
    class="emotion-0 emotion-1"
  >
    Dummy text
  </span>
</div>
`;
